### PR TITLE
added typings for vmap-js

### PR DESCRIPTION
**Pull request by jgainfort**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

Typings are for https://github.com/dailymotion/vmap-js

### CODE REVIEW

**Comment by jgainfort** [comment]
Reviewer:

I want to change the index.d.ts to the following but I am not sure if this goes against spec.

```
// Type definitions for vmap 2.2
// Project: https://github.com/dailymotion/vmap-js#readme
// Definitions by: John G. Gainfort, Jr. <https://github.com/me>
// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
// TypeScript Version: 2.7

declare namespace VMAP {
    /**
     * Provides information about an ad break
     */
    interface VMAPAdBreak {
        /**
         * Represents the timing of the ad break
         */
        timeOffset: string;
        /**
         * Identifies whether the ad break allows "linear", "nonlinear" or "display" ads
         */
        breakType: 'linear' | 'nonlinear' | 'display';
        /**
         * An optional string identifier for the ad break
         */
        breakId?: string;
        /**
         * An option used to distribute ad breaks equally spaced apart from one another along a linear timeline
         */
        repeatAfter: string;
        /**
         * An array of VMAPTrackingEvent objects with tracking URLs
         */
        trackingEvents: VMAPTrackingEvent[];
        /**
         * An array of VMAPExtension objects
         */
        extensions: VMAPExtension[];
        /**
         * A VMAPAdSource object
         */
        adSource: VMAPAdSource;
        /**
         * Call the trackers for the given event with an option error code parameter for error events
         */
        track: (event: string, errorCode?: string) => void;
    }

    /**
     * Provides information about a VMAP tracking event
     */
    interface VMAPTrackingEvent {
        /**
         * The name of the event to track for the element. Can be one of breakStart, breakEnd or error
         */
        event: 'breakStart' | 'breakEnd' | 'error';
        /**
         * The URI of the tracker
         */
        uri: string;
    }

    /**
     * Provides the player with either an inline ad response or a reference to an ad response
     */
    interface VMAPAdSource {
        /**
         * Ad identifier for the ad source
         */
        id: string;
        /**
         * Indicates whether a VAST ad pod or multple buffet of ads can be served into an ad break
         */
        allowMultipleAds: boolean;
        /**
         * Indicates whether the video player should honor the redirects within an ad response
         */
        followRedirects: boolean;
        /**
         * Contains a URI to the VAST
         */
        adTagURI: string;
        /**
         * Contains custom ad data
         */
        customData: any;
        /**
         * Contains an embedded VAST response
         */
        vastAdData: Element;
    }

    /**
     * Provides information about VMAP extension
     */
    interface VMAPExtension {
        /**
         * Object containing all this extension children and their name as the key
         */
        children: Object;
        /**
         * Object containing all this extension attributes and their name as the key
         */
        attribute: Object;
        /**
         * Object parsed from CDATA or as a fallback all of the text nodes of this extension concatenated
         */
        value: Object;
    }
}

/**
 * VMAP parser
 */
declare class VMAP {
    /**
     * The VMAP version (should be 1.0)
     */
    version: string;
    /**
     * An array of VMAPAdBreak objects
     */
    adBreaks: VMAP.VMAPAdBreak[];
    /**
     * An array of VMAPExtension objects
     */
    extensions: VMAP.VMAPExtension[];

    constructor(xml: Document);
}

export as namespace VMAP;
export = VMAP;
```

The reason being so I can do something like

```
import VMAP from 'vmap'

extension MyVMAPAdSource extends VMAP.VMAPADSource { 
    addedProperty: any,
    ...
}
```

As it stands now I can only access the single VMAP class.

If this is valid I can commit immediately. Please let me know!

File: types/vmap/tslint.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{ "extends": "dtslint/dt.json",
+  "rules": {
```
Comment (jessetrinity) [inline types/vmap/tslint.json]: You should only need `{ "extends": "dtslint/dt.json" }`

**Review by jessetrinity — CHANGES_REQUESTED** [review]

File: types/vmap/tslint.json (hunk 1 above)
Comment (jgainfort) [reply types/vmap/tslint.json]: ban-types is due to this interface

The library maintainer does not specify what these are.

```
interface VMAPExtension {
    /**
     * Object containing all this extension children and their name as the key
     */
    children: Object;
    /**
     * Object containing all this extension attributes and their name as the key
     */
    attribute: Object;
    /**
     * Object parsed from CDATA or as a fallback all of the text nodes of this extension concatenated
     */
    value: Object;
}```

Comment (jessetrinity) [reply types/vmap/tslint.json]: Can you use `any`, or if you know it is an object, use `object`?

Comment (jgainfort) [reply types/vmap/tslint.json]: done

**Comment by jessetrinity** [comment]
>I want to change the index.d.ts to the following but I am not sure if this goes against spec.

I think you want to check out [this page](https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-class-d-ts.html), in particular the part at the bottom about exposing types. Also note the bit about not needing `export as namespace` if this is not a UMD module.

Then I think you should be able to do
```typescript
import VMAP, { VMAPAdSource } from 'vmap';
```

**Comment by jgainfort** [comment]
@jessetrinity Thank you for the pointer!

**Pull request merged**